satellite/satellitedb/users.go

```diff
@@ -48,9 +48,9 @@ func (users *users) GetByEmail(ctx context.Context, email string) (_ *console.Us
 // Insert is a method for inserting user into the database
 func (users *users) Insert(ctx context.Context, user *console.User) (_ *console.User, err error) {
 	defer mon.Task()(&ctx)(&err)
-	userID, err := uuid.New()
-	if err != nil {
-		return nil, err
+
+	if user.ID.IsZero() {
```

Comment:
do other fields need to be checked for nil values?

Reply:
I think the caller checks them atm

Reply:
that's is not enough to check if it's valid
`func New() (*UUID, error) {
	uuid := new(UUID)

	n, err := io.ReadFull(rand.Reader, uuid[:])
	if err != nil {
		return nil, err
	} else if n != len(uuid) {
		return nil, errors.New(fmt.Sprintf("insufficient random data (expected: %d, read: %d)", len(uuid), n))
	} else {
		// variant bits; for more info
		// see https://www.ietf.org/rfc/rfc4122.txt section 4.1.1
		uuid[8] = uuid[8]&0x3f | 0x80
		// version 4 (pseudo-random); for more info
		// see https://www.ietf.org/rfc/rfc4122.txt section 4.1.3
		uuid[6] = uuid[6]&0x0f | 0x40
	}

	return uuid, nil
}
`

Reply:
If we implement the random generation at a lower level (in `CreateUser` in `service.go`), I don't think randomness validation is necessary since we will be sure of how it is being generated.